openmc/plotter.py

```diff
@@ -0,0 +1,617 @@
+from numbers import Integral, Real
+
+import numpy as np
+
+import openmc.checkvalue as cv
+import openmc.data
+
+# Supported keywords for xs plotting
+PLOT_TYPES = ['total', 'scatter', 'elastic', 'inelastic', 'fission',
+              'absorption', 'capture', 'nu-fission', 'nu-scatter', 'unity',
+              'slowing-down power', 'damage']
+
+# Special MT values
+UNITY_MT = -1
+XI_MT = -2
+
+# MTs to combine to generate associated plot_types
+PLOT_TYPES_MT = {'total': (2, 3,),
+                 'scatter': (2, 4, 11, 16, 17, 22, 23, 24, 25, 28, 29, 30,
+                             32, 33, 34, 35, 36, 37, 41, 42, 44, 45, 152,
+                             153, 154, 156, 157, 158, 159, 160, 161, 162,
+                             163, 164, 165, 166, 167, 168, 169, 170, 171,
+                             172, 173, 174, 175, 176, 177, 178, 179, 180,
+                             181, 183, 184, 190, 194, 196, 198, 199, 200,
+                             875, 891),
```

Comment:
A more rigorous way to handle this would be to use openmc.data.endf.SUM_RULES (e.g., look at `openmc.data.IncidentNeutron.get_reaction_components()`). Can you look into refactoring?